client/lib/browser-storage/index.ts

```diff
@@ -32,7 +34,15 @@ const getDB = once( () => {
 					}
 					reject( request.error );
 				};
-				request.onsuccess = () => resolve( request.result );
+				request.onsuccess = () => {
+					const db = request.result;
+					db.onerror = function( errorEvent: any ) {
```

Comment:
This line is very surprising. You're adding a handler to `db.error` inside the handler for a request success. Can a DB error ever happen if the request is successful? And is there a difference vs adding this code to the `request.onerror` handler?

Reply:
I'm adding an error handler to the `db`, not _the request to open the db_. I agree it looks a little wild. :) This is a generic error handler that will get called whenever an error occurs on any future db operation using this handle due to how error propagation works in idb.

Reply:
see https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#Handling_Errors

Reply:
Added this is a code comment

Reply:
Oh, I see! So you're only registering the error handler once the first request goes through. And since this only happens once, it all works out too. That makes sense, thanks! 👍

Reply:
Right, and even if we change this later to use a different strategy to manage db handles, it still works. You just attach one error handler per db handle.